### PR TITLE
#55 Implement search in Grid and List modes

### PR DESCRIPTION
**Pull request by dialexo**
From #55 this PR implements client-side local search functionality for Grid and List display modes:

Peek 2021-07-05 19-33

Also improves grid header display to be consistent with table view header:

Before | After
--- | ---
Screenshot from 2021-07-05 19-36-29 |  Screenshot from 2021-07-05 19-35-03

### Summary

- implements `useItemSearch` hook, which manages search state and returns search results + `ItemSearchInput` component to render the text input
- forward this `ItemSearchInput` through `ItemsGrid` and `ItemsTable` to `TableToolbar`
- harmonizes table and grid headers to use `TableToolbar` in both cases, add padding to `EmptyItem`
- adds integration tests for search

### CODE REVIEW

**Comment by dialexo** [comment]
Thanks a lot! Both issues have been fixed :+1:

File: src/components/main/TableToolbar.js [hunk 1]
```diff
@@ -57,6 +58,8 @@ const TableToolbar = (props) => {
         </Typography>
       )}
 
+      {numSelected > 0 ? null : itemSearchInput}
```
Comment (pyphilia) [inline src/components/main/TableToolbar.js:61]: Would the following work? `!numSelected  &&  itemSearchInput`
Perhaps you can even merge with the following condition.

File: src/langs/fr.json [hunk 2]
```diff
@@ -85,6 +85,9 @@
     "Profile": "Profil",
     "Member ID": "ID de Membre",
     "Size": "Taille",
-    "Manage Access": "Gérer les accès"
+    "Manage Access": "Gérer les accès",
+    "This item is empty.": "Cet élément est vide.",
+    "Search…": "Recherche…",
+    "No search results found.": "Aucun résultat correspondant à la recherche."
```
Comment (pyphilia) [inline src/langs/fr.json:91]: `Aucun résultat ne correspond à la recherche`?

**Review by pyphilia — CHANGES_REQUESTED** [review]
Nice! I really like how you factor out the search bar. I left some minor comments,

**Comment by dialexo** [comment]
Thanks! I've pushed both changes

**Review by pyphilia — APPROVED** [review]

**Pull request merged**